internal/langserver/handlers/command/module_calls.go

```diff
@@ -89,11 +94,16 @@ func parseModuleRecords(records []datadir.ModuleRecord) []moduleCall {
 			subModuleName = v[1]
 		}
 
+		docsLink, err := getModuleDocumentationLink(ctx, manifest)
+		if err != nil {
+			return nil, err
```

Comment:
The `docsURL` function throws if we can't parse the URL, but it's a URL we created and control. `url.parse` only throws if the URL we create is syntactically invalid. I'm missing where this would be a useful thing to do.

My main concern is if we return an error here, we miss all the other relevant module call info we built. The client side views won't have information to display and/or possibly a large error message for n+1 module calls.

Reply:
The error is relatively unlikely to be returned - it would involve an invalid source address in config, but I addressed this a separate commit which logs the error and leaves the link empty, so we can still render it in the UI without the link.

FYI there's also https://github.com/hashicorp/terraform-ls/issues/906